depresolve.go

```diff
@@ -214,3 +210,24 @@ func standardRepoHostImportPathToCloneURL(importPath string) (string, error) {
 	}
 	return "https://" + strings.Join(parts[:3], "/") + ".git", nil
 }
+
+// replaceImportPathRepoRoot modifies the given importPath by replacing the
+// root string in the importPath with the host+path of the clone URL.
+// This is necessary for resolving refs to custom import path packages, since
+// the defs within those packages would have the Unit field set to
+// "${repoURI}/path/to/pkg/dir", where repoURI is the host+path of the cloneURL.
+//
+// This is a HACK to make def resolution work in presence of custom import
+// paths. A proper solution would require passing in the custom import path
+// information from srclib to the graph step, and setting that as the Unit field
+// of the defs identified in the source unit.
+//
+// TODO: Implement the proper solution and get rid of this hack.
+func replaceImportPathRepoRoot(importPath, root, cloneURL string) string {
+	i := strings.Index(cloneURL, "://")
+	if i < 0 {
+		return importPath
+	}
+	newRoot := strings.TrimSuffix(cloneURL[i+len("://"):], ".git")
```

Comment:
I would be very careful about doing `strings.TrimSuffix(..., ".git")`. I understand you're doing it to normalize clone urls such as "https://github.com/user/repo.git" and "https://github.com/user/repo", because on _most_ Git servers, the both are supported.

However, I've seen some git servers where that's not the case, and the .git suffix cannot be trimmed optionally.

So, if possible, I suggest you trim this .git suffix at a higher level where it can be done without having to guess if it's an optional suffix or an important part of the clone URL.

That said, this is pretty minor and only going to affect some small number of repos (less than 10%?). So if fixing this properly is too expensive now, it might be fine to defer fixing this until later.